src/iceberg/catalog/rest/endpoint.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <string_view>
+
+#include "iceberg/catalog/rest/iceberg_rest_export.h"
+#include "iceberg/result.h"
+
+/// \file iceberg/catalog/rest/endpoint.h
+/// Endpoint definitions for Iceberg REST API operations.
+
+namespace iceberg::rest {
+
+/// \brief HTTP method enumeration.
+enum class HttpMethod : uint8_t { GET, POST, PUT, DELETE_, HEAD };
+
+/// \brief Convert HttpMethod to string representation.
+constexpr std::string_view ToString(HttpMethod method);
+
+/// \brief An Endpoint is an immutable value object identifying a specific REST API
+/// operation. It consists of:
+/// - HTTP method (GET, POST, DELETE, etc.)
+/// - Path template (e.g., "/v1/{prefix}/namespaces/{namespace}")
+class ICEBERG_REST_EXPORT Endpoint {
+ public:
+  /// \brief Create an endpoint with method and path template.
+  ///
+  /// \param method HTTP method (GET, POST, etc.)
+  /// \param path_template Path template with placeholders (e.g., "/v1/{prefix}/tables")
+  /// \return Endpoint instance or error if invalid
+  static Result<Endpoint> Create(HttpMethod method, std::string path_template);
+
+  /// \brief Parse endpoint from string representation.
+  ///
+  /// \param str String in format "METHOD /path/template" (e.g., "GET /v1/namespaces")
+  /// \return Endpoint instance or error if malformed
+  static Result<Endpoint> FromString(std::string_view str);
+
+  /// \brief Get the HTTP method.
+  constexpr HttpMethod method() const { return method_; }
+
+  /// \brief Get the path template.
+  std::string_view path_template() const { return path_template_; }
```

Comment:
```suggestion
  /// \brief Get the path.
  std::string_view path() const { return path_; }
```

Let's make it shorter.